lib/resolveurl/plugins/anistream.py

```diff
@@ -1,24 +1,44 @@
 """
-ani-stream resolveurl plugin
-Copyright (C) 2016 quartoxuna
+    Plugin for ResolveUrl
+    Copyright (C) 2020 gujal
 
-This program is free software: you can redistribute it and/or modify
-it under the terms of the GNU General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
 
-This program is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-GNU General Public License for more details.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
 
-You should have received a copy of the GNU General Public License
-along with this program. If not, see <http://www.gnu.org/licenses/>.
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-from resolveurl.plugins.__resolve_generic__ import ResolveGeneric
 
+from resolveurl.plugins.lib import helpers
+import re, base64
+from resolveurl import common
+from resolveurl.resolver import ResolveUrl, ResolverError
 
-class AniStreamResolver(ResolveGeneric):
+
+class AniStreamResolver(ResolveUrl):
     name = "ani-stream"
     domains = ["ani-stream.com"]
     pattern = r'(?://|\.)(ani-stream\.com)/(?:embed-)?([0-9a-zA-Z-]+)'
+
+
+    def get_media_url(self, host, media_id):
+        web_url = self.get_url(host, media_id)
+        headers = {'User-Agent': common.FF_USER_AGENT}
+        html = self.net.http_GET(web_url, headers=headers).content
+        r = re.search(r'base64,([^"]+)',html)
+        if r:
+            html = base64.b64decode(r.group(1)).decode('utf-8')
```

Comment:
Please test this in Kodi 19 as the current code will cause issue in Python3
specifically the decode and base64 have to be handled correctly for Python3